code/modules/admin/bccm/bccm.dm

```diff
@@ -0,0 +1,387 @@
+// BCCM (Ban Counter Counter Measures system) ((Name subject to change)), originally inspired by EAMS (Epic Anti-Multiaccount System), by Epicus
+//version 1.0.2
+
+/datum/bccm_info
+	var/is_loaded = FALSE
+	var/is_whitelisted = FALSE
+
+	var/ip
+	var/ip_as
+	var/ip_mobile
+	var/ip_proxy
+	var/ip_hosting
+
+/client
+	var/datum/bccm_info/bccm_info = new
+
+SUBSYSTEM_DEF(bccm)
+	name = "BCCM"
+	init_order = SS_INIT_BCCM
+	flags = SS_NO_FIRE
+
+	var/max_error_count = 4
+
+	var/is_active = FALSE
+	var/error_counter = 0
+
+	var/list/tgui_panel_asn_data = list()
+	var/list/tgui_panel_wl_data = list()
+
+	var/list/client/postponed_client_queue = new
+
+/datum/controller/subsystem/bccm/Initialize(timeofday)
+	if(!config.bccm)
+		return ..()
+
+	if(!sqlenabled)
+		log_debug("BCCM could not be loaded without SQL enabled")
+		return ..()
+
+	Toggle()
+	return ..()
+
+/datum/controller/subsystem/bccm/stat_entry(msg)
+	if(is_active)
+		msg = "ACTIVE"
+	else
+		msg = "OFFLINE"
+	return msg
+
+/datum/controller/subsystem/bccm/proc/Toggle(mob/user)
+	if (!initialized && user)
+		return
+
+	if(!is_active && !SSdbcore.Connect())
+		log_debug("BCCM could not be loaded because the DB connection could not be established.")
+		return
+
+	is_active = !is_active
+	log_debug("BCCM is [is_active ? "enabled" : "disabled"]!")
+	if(is_active)
+
+		tgui_panel_asn_data = GetAsnBanlistDatabase()
+		tgui_panel_wl_data = GetWhitelistDatabase()
+
+		var/list/clients_to_check = postponed_client_queue.Copy()
+		postponed_client_queue.Cut()
+		for (var/client/C in clients_to_check)
+			CollectClientData(C)
+			HandleClientAccessCheck(C, postponed = TRUE)
+			HandleASNbanCheck(C, postponed = TRUE)
+			CHECK_TICK
+	return is_active
```

Comment:
```suggestion
	. = is_active
	if(.)
		return

	tgui_panel_asn_data = GetAsnBanlistDatabase()
	tgui_panel_wl_data = GetWhitelistDatabase()

	var/list/clients_to_check = postponed_client_queue.Copy()
	postponed_client_queue.Cut()
	for (var/client/C in clients_to_check)
		CollectClientData(C)
		HandleClientAccessCheck(C, postponed = TRUE)
		HandleASNbanCheck(C, postponed = TRUE)
		CHECK_TICK
```
this is just a "this looks prettier and should work the same" suggestion that you can frankly ignore but less indentation = prettier = more readable in my eyes